app/src/main/java/com/example/cardgame/GameplayFragment.kt

```diff
@@ -335,69 +217,105 @@ class GameplayFragment : Fragment() {
+        disposable?.dispose()
+        disposable = getCardObservable()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe { cardInfo ->
 
-        client.newCall(request).enqueue(object : Callback {
-            override fun onFailure(call: Call, e: IOException) {
-                e.printStackTrace()
-                Log.d(TAG, "REQUEST FOR NEW CARD FAILED!", e)
-            }
+                cardInfo?.apply {
+
+                    currentImage = image
+                    returnCard = Consts.indexOf(value)
 
-            override fun onResponse(call: Call, response: Response) {
-                if(response.isSuccessful) {
-                    val result = response.body()!!.string()
-                    val strRes = Gson().fromJson<ResponseForNewCard>(result,
-                        ResponseForNewCard::class.java)
-                    val image = strRes.cards[0].image
-                    val cardValue = strRes.cards[0].value
-                    returnCard = Consts.indexOf(cardValue)
                     activity?.runOnUiThread {
-                        val imgView = root.findViewById<ImageView>(R.id.ivCurrentCard)
+
+                        val imgBack = root!!.findViewById<ImageView>(R.id.ivCurrentCardBack)
+                        val imgView = root!!.findViewById<ImageView>(R.id.ivCurrentCard)
+                        val easyFlipView = root!!.findViewById<EasyFlipView>(R.id.easyFlipView)
+
+                        Picasso.get().load(R.drawable.back_of_a_card).into(imgBack)
+
+                        easyFlipView.flipDuration = 500
+                        easyFlipView.setFlipTypeFromLeft()
+                        easyFlipView.flipTheView(true)
+
+                        easyFlipView.flipDuration = 2000
+                        easyFlipView.setFlipTypeFromLeft()
+                        easyFlipView.flipTheView(true)
+
                         Picasso.get().load(image).into(imgView)
+
+                        val tv = root!!.findViewById<TextView>(R.id.tvScore)
+                        tv.text = currentScore.toString()
                     }
-                }
-                else {
-                    Log.d(TAG, "RESPONSE FOR NEW CARD IS UNSUCCESSFUL")
+
+                    val nextCard = returnCard
+                    if (shouldCompare) {
+                        if (op(nextCard, currentCard)) {
+                            if (currentCard > nextCard)
+                                Log.d(
+                                    TAG,
+                                    "${Consts.getName(nextCard)} < ${Consts.getName(currentCard)}"
+                                )
+                            else
+                                Log.d(
+                                    TAG,
+                                    "${Consts.getName(nextCard)} > ${Consts.getName(currentCard)}"
+                                )
```

Comment:
The `op` function already does the comparison between the current and next card so the extra if is redundant. If you wish to log the outcome put it in a function that comapres the values and then logs them and pass that function to `drawNewCard()` instead of the lambda.